x-pack/plugins/security_solution/public/flyout/url/expandable_flyout_state_from_event_meta.ts

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ExpandableFlyoutContext } from '@kbn/expandable-flyout';
+import { RightPanelKey } from '../right';
+
+interface RedirectParams {
+  index: string;
+  eventId: string;
+  scopeId: string;
+}
+
+/**
+ * Builds flyout state from basic event-related data, such as index name, event id and scope id.
+ * This value can be used to open the flyout either by passing it directly to the flyout api (exposed via ref) or
+ * by serializing it to the url & performing a redirect
+ */
+export const expandableFlyoutStateFromEventMeta = ({
+  index,
+  eventId,
+  scopeId,
+}: RedirectParams): ExpandableFlyoutContext['panels'] => {
+  return {
+    right: {
+      id: RightPanelKey,
+      params: {
+        id: eventId,
+        indexName: index,
+        scopeId,
+      },
+    },
+    left: undefined,
```

Comment:
this makes me thing that we wouldn't save the left section state when using the share button. Is that by design? I think we do want to save the entire state of the flyout.

The biggest question I have is what's the best way to generate alerts that have a correct `kibana.alert.url` value? Using the `yarn test:generate` script and the default Elastic rules that field is empty for me...

Reply:
share button will use **whatever you currently have serialized** as `eventFlyout` query parameter, and that includes the entire state - and that will be appended to `kibana.alert.url` & fowarded correctly to the flyout. that is what I needed that clipboard magic for. :)

this bit handles the case where user comes in from contexts other than initiated by clicking share button. we could provide the exact state information somehow - but I imagine it as having some simplified parameter like '?threats', just as discussed with our PM - that was considered optional and non prio for now. @PhilippeOberti

Reply:
that makes sense, thanks for the clarification!